clarity_ext/service/file_service.py

```diff
@@ -122,3 +127,40 @@ def cleanup(self):
 class SharedFileNotFound(Exception):
     pass
 
+
+class Csv:
+    """A simple wrapper for csv files"""
+    def __init__(self, file, delim=","):
+        lines = list()
+        for line in file:
+            values = line.strip().split(delim)
+            csv_line = CsvLine(values, self)
+            lines.append(CsvLine(values, self))
+            if len(lines) == 1:
```

Comment:
When I wrote it I was sure it would be commented on, since I totally agree. Perhaps I should have commented on this.

It was like you're recommending at first, but since CsvLine requires that self.key_to_index in this object has been created already (circular dependency) I did it like this - rather than breaking up that dependency somehow. Decided to go with the hacky/non-readable solution in this case.